Consolidate Reference docs section contents

ptgott opened the pull request:
Closes #56206

Reduce the number of first-level children in the Reference section of the docs from 23 to 9, not counting the index page available at `/docs/reference/`. This makes the Reference section easier to navigate, since a user can more readily form a mental model of the content available in the section with fewer items and a more thematic organization.

- Add a section for deployment references, consolidating reference guides for configuring and architecting a Teleport deployment.
- Move the Predicate Language and User Types references into the Access Controls section of References.
- Add a section for API reference guides, consolidating the Terraform, MWI Terraform, and Operator sections.
- Move Cloud FAQ to Quick Links. This places the guide next to the main FAQ in the sidebar to make it more discoverable, since the guide is currently rather hidden. This also helps pave the way to merge the two FAQs in the future.

ptgott committed:
Consolidate Reference docs section contents

Closes #56206

Reduce the number of first-level children in the Reference section of
the docs from 23 to 9, not counting the index page available at
`/docs/reference/`. This makes the Reference section easier to navigate,
since a user can more readily form a mental model of the content
available in the section with fewer items and a more thematic
organization.

- Add a section for deployment references, consolidating reference
  guides for configuring and architecting a Teleport deployment.
- Move the Predicate Language and User Types references into the Access
  Controls section of References.
- Add a section for API reference guides, consolidating the Terraform,
  MWI Terraform, and Operator sections.
- Move Cloud FAQ to Quick Links. This places the guide next to the main
  FAQ in the sidebar to make it more discoverable, since the guide is
  currently rather hidden. This also helps pave the way to merge the two
  FAQs in the future.

strideynet commented on integrations/terraform-mwi/gen/docs.sh:
```diff
@@ -21,7 +21,7 @@ if [ -z "$VERSION" ]; then
 
 TFDIR="$(pwd)"
-DOCSDIR="$(pwd)/../../docs/pages/reference/terraform-provider-mwi"
+DOCSDIR="$(pwd)/../../docs/pages/reference/api/terraform-provider-mwi"
```
Yes - that sounds good - I also considered moving that entire reference section under the MWI section.

ptgott committed:
Add an MWI reference section

The MWI Terraform module reference does not belong with reference guides
for API clients. Add a section to the reference guide for Machine &
Workload Identity.

ravicious reviewed (approved):
LGTM, I definitely thought so too that there are too many pages under the "References" section.

ptgott committed:
Rename "API" to "Infrastructure as Code"

Teleport users expect tctl, Operator, and Terraform Provider docs to be
in an Infrastructure as Code reference section, rather than the more
vague "API" section.

Responds to **ravicious** feedback.
